src/write.c

```diff
@@ -590,38 +600,41 @@ static void avifEncoderWriteColorProperties(avifRWStream * outputStream,
     if (dedup) {
         avifItemPropertyDedupStart(dedup);
     }
-    avifBoxMarker colr = avifRWStreamWriteBox(dedupStream, "colr", AVIF_BOX_SIZE_TBD);
-    avifRWStreamWriteChars(dedupStream, "nclx", 4);                            // unsigned int(32) colour_type;
-    avifRWStreamWriteU16(dedupStream, imageMetadata->colorPrimaries);          // unsigned int(16) colour_primaries;
-    avifRWStreamWriteU16(dedupStream, imageMetadata->transferCharacteristics); // unsigned int(16) transfer_characteristics;
-    avifRWStreamWriteU16(dedupStream, imageMetadata->matrixCoefficients);      // unsigned int(16) matrix_coefficients;
-    avifRWStreamWriteU8(dedupStream, (imageMetadata->yuvRange == AVIF_RANGE_FULL) ? 0x80 : 0); // unsigned int(1) full_range_flag;
-                                                                                               // unsigned int(7) reserved = 0;
+    avifBoxMarker colr;
+    AVIF_CHECKRES(avifRWStreamWriteBox(dedupStream, "colr", AVIF_BOX_SIZE_TBD, &colr));
+    AVIF_CHECKRES(avifRWStreamWriteChars(dedupStream, "nclx", 4));                   // unsigned int(32) colour_type;
+    AVIF_CHECKRES(avifRWStreamWriteU16(dedupStream, imageMetadata->colorPrimaries)); // unsigned int(16) colour_primaries;
+    AVIF_CHECKRES(avifRWStreamWriteU16(dedupStream, imageMetadata->transferCharacteristics)); // unsigned int(16) transfer_characteristics;
+    AVIF_CHECKRES(avifRWStreamWriteU16(dedupStream, imageMetadata->matrixCoefficients)); // unsigned int(16) matrix_coefficients;
+    AVIF_CHECKRES(avifRWStreamWriteU8(dedupStream, (imageMetadata->yuvRange == AVIF_RANGE_FULL) ? 0x80 : 0)); // unsigned int(1) full_range_flag;
+    // unsigned int(7) reserved = 0;
     avifRWStreamFinishBox(dedupStream, colr);
     if (dedup) {
-        ipmaPush(ipma, avifItemPropertyDedupFinish(dedup, outputStream), AVIF_FALSE);
+        AVIF_CHECKRES(avifItemPropertyDedupFinishAndIpmaPush(dedup, outputStream, ipma, AVIF_FALSE));
     }
 
-    avifEncoderWriteExtendedColorProperties(dedupStream, outputStream, imageMetadata, ipma, dedup);
+    AVIF_CHECKRES(avifEncoderWriteExtendedColorProperties(dedupStream, outputStream, imageMetadata, ipma, dedup));
+    return AVIF_RESULT_OK;
```

Comment:
Nit: Replace these two lines (616-617) with `return avifEncoderWriteExtendedColorProperties(...);`

This is okay if you will be adding code between these two lines.